Initial PR for Fundamentals of Computer Vision (#73)
johko/computer-vision-course: chapters/en/Unit 1 - Fundamentals/Chapter 1 - Fundamentals of Computer Vision/defination.mdx

# What Is Computer Vision

Let's revisit our example from the previous chapter: kicking a ball. As we have seen,
this involves multiple tasks that our brains can do in a split second. The extraction of
what is meaningful information from an image input is at the core of computer vision.
But what is computer vision?

## Definition
Computer vision is a multi-disciplinary subfield of artificial intelligence (AI)  
that enables computer systems to analyze, interpret, and process visual information.
It involves the development of algorithms, models, and systems that can generate images or extract knowledge from images,

> videos, and other forms of visual data.

Author: It's an iteration of a text book definition, youtube videos and web searches but i see you point, we will try to update this with a straight up textbook definition.

from the hard-coded rules, but it still relies on domain knowledge and exhaustive feature engineering.
The competing strategy arises when deep learning methods and large datasets meet.
Deep learning (DL) allows machines to learn complex features from the raw data automatically. This paradigm
shift allowed us to build more adaptive and sophisticated models causing a renaissance in the field.

We wanted to emphasize here that the compute vision community is made up of researchers,
engineers, data scientists, and passionate hobbyists across the globe coming from a vast array
of disciplines. This community maintains the Open-source initiates, leads community-driven projects,
and democratizes access to cutting-edge tools and technologies.

## Interdisciplinary with other fields + Image Understanding

Just as it is hard to draw a line that separates artificial intelligence and computer vision,
it is also hard to separate computer vision from its neighbouring fields. Take image preprocessing and analysis
as an example. A tentative separation is that the input and output of image analysis are always images.

> However, this is a myopic take. Even simple tasks, such as calculating the medium value of an image, would

Reviewer: ```suggestion
However, this is a shortsighted take. Even simple tasks, such as calculating the medium value of an image, would
```
I think we should try to keep the language relatively simple, as the audience of the course will not necessarily be academics.

# What Is Computer Vision

Let's revisit our example from the previous chapter: kicking a ball. As we have seen,
this involves multiple tasks that our brains can do in a split second. The extraction of
what is meaningful information from an image input is at the core of computer vision.

> But what precisely is computer vision?

Reviewer 2: ```suggestion
But what is computer vision?
```

## Definition

> Computer Vision is a multi-disciplinary field of Artificial intelligence (AI)  

Reviewer 2: ```suggestion
Computer vision is a multi-disciplinary subfield of artificial intelligence (AI)  
```

that enables computer systems to analyze, interpret, and process visual information.

> It involves the development of algorithms, models, and systems that can extract knowledge from images,

Reviewer 2: ```suggestion
It involves the development of algorithms, models, and systems that can generate images or extract knowledge from images,
```

videos, and other forms of visual data.

> 

Reviewer 2: ```suggestion
```

## Deep learning and the computer vision renaissance

> 

Reviewer 2: ```suggestion
```

The evolution of computer vision has been marked by a series of incremental advancements in
and across its interdisciplinary fields, where each step forward gave rise to breakthrough algorithms,
hardware, and data, giving it more power and flexibility. One such history was the jump to the widespread

> use of deep learning methods.

Reviewer 2: This sentence was a bit too long and there are newlines, it's better if we don't have any newlines where there shouldn't be

Author: Actually, the new line characters and spacings you see across definition.mdx were kept as i thought they might be easy to read like that but i was clearly confused on how mdx files are render. I'll redo these spacing adjustments along with changes mentioned in upcoming patch 🤗

> Initially, to extract and learn the information in an image you extract features through image-preprocessing

Reviewer 2: ```suggestion
Initially, to extract and learn the information in an image, you extract features through image-preprocessing
```

techniques (chapter 3). Once you have a group of features describing your image, you use a classical
machine learning algorithm for your dataset of features. It is a strategy that already simplifies things

> from the hard-code rules, but it still relies on domain knowledge and exhaustive feature engineering.

Reviewer 2: ```suggestion
from the hard-coded rules, but it still relies on domain knowledge and exhaustive feature engineering.
```

> The competing strategy arises when deep learning methods and large datasets meet each other.

Reviewer 2: ```suggestion
The competing strategy arises when deep learning methods and large datasets meet.
```

> Deep Learning(DL) allowed machines to automatically learn complex features from the raw data. This paradigm

Reviewer 2: ```suggestion
Deep learning (DL) allows machines to learn complex features from the raw data automatically. This paradigm
```

shift allowed us to build more adaptive and sophisticated models causing a renaissance in the field.

> We wanted to emphasize here that the compute vision community is made up of researchers,

Reviewer 3: Parts like this seem like they provide notes for reviewers, not a part of the content, perhaps it should be separated from the main content as a comment, so it's also easier to clean up later. 

e. g. 
```
<!--
Note: we wanted to emphasize ....
-->
```

engineers, data scientists, and passionate hobbyists across the globe coming from a vast array
of disciplines. This community maintains the Open-source initiates, leads community-driven projects,
and democratizes access to cutting-edge tools and technologies.

## Interdisciplinary with other fields + Image Understanding

  
Just as it is hard to draw a line that separates artificial intelligence and computer vision,
it is also hard to separate computer vision from its neighbouring fields. Take image preprocessing and analysis
as an example. A tentative separation is that the input and output of image analysis are always images.
However, this is a myopic take. Even simple tasks, such as calculating the medium value of an image, would
be classified under computer vision. To clarify their differences, we must introduce a new concept of image
understanding.

Image understanding is the process of making sense of the content of an image. It can be defined in three

> different levels :

Reviewer 2: ```suggestion
different levels:
```

**Low-level processes** are primitive operations on images (i.e. image sharpening, changing the contrast).
The input and the output are images. **Mid-level processes** include segmentation, description of objects,
and object classification. The information is an image, but the result is attributes associated with the image.
This could be done with a combination of image preprocessing and ML algorithms.

> 

Reviewer 2: ```suggestion
```

**High-level processes** include making sense of the entirety of an image, i.e., recognition of a given object,
scene reconstruction, and image-to-text. These are tasks typically associated with human cognition.

Image analysis is mainly concerned with low and mid-level processes. However, computer vision is interested
in mid- and high-level processes. Thus, there is an overlap in the mid-level processes between image analysis
and computer vision.

It is essential to remember this since allocating resources to develop a sophisticated model, such as a
neuronal network for data-poor scenarios or simple images, might not be appropriate. From a business point
of view, model development costs time and money; it is necessary to know when to use the right tools.

> Combining a “preprocessing” part before moving on to a more robust model is not unusual.

Reviewer 2: ```suggestion
Combining a “preprocessing” part before moving on to a more robust model is usual.
```

On the opposite side, sometimes, the layers of a neural network automatically perform tasks like these,
eliminating the need for explicit preprocessing. Image analysis might act as a first exploratory data analysis
step for those familiar with data science. Lastly, classical image analysis methods can also be used for
data augmentation to improve the quality and diversity of training data for computer vision models.

> NOTE(TO HF TEAM): The following sections are not part of the initial draft but somehow we felt they might be

Reviewer 2: ♥️ 🤗

useful and looking to iterate based on feedback.

## Computer Vision Tasks Overview

> We have seen before that this is a really hard task because computers have no previous knowledge of the world.

Reviewer 3: A bit of a stylistic mismatch here, the title is "tasks" (plural) and in the first sentence you talk about "a task". Probably better to rephrase. E.g. "We have seen before that computer vision is really hard for computers because they have no previous knowledge of the world."

In our example, we start knowing what a ball is, how to track its movement, how objects usually move in space,
how to estimate when the ball will reach us, where is your foot, how a foot moves, and how to estimate how much
force you need to hit the ball. If we were to break this down into specific computer vision tasks,
we would have:

  - Scene Recognition 
  - Object Recognition
  - Object Detection
  - Segmentation (instance, semantic)
  - Tracking 
  - Dynamic Environment Adaptation
  - Path Planning

> You will see more about the core tasks of computer vision in the [CV TASKS] chapter.

Reviewer 3: ```suggestion
You will read more about the core tasks of computer vision in the [CV TASKS] chapter.
```

significantly based on factors such as lighting conditions, the presence of occlusions, the resolution
of the image, and the quality of the camera. In low-light conditions or with pixelated images, even the
seemingly basic task of pedestrian detection can become exceedingly complex for computer vision algorithms,
requiring advanced image enhancement and machine learning techniques. Therefore, the challenge in image analysis
and computer vision often lies not in the inherent nobility of a task, but in the intricacies of the visual data
and the computational methods required to extract meaningful insights from it.

## Link to computer vision applications

As a field, computer vision has a growing importance in society. There are many ethical considerations
regarding its applications. For example, a model that is deployed to detect cancer can have terrible
consequences if it classifies a cancer sample as healthy. Surveillance technology, such as models that
are capable of tracking people, also raises a lot of privacy concerns. This will be discussed in detail
in Chapter 14- Applications of Computer Vision and real-world Use Cases, but we will give you a taste of

> some of its application

Reviewer 3: ```suggestion
some of its applications.
```

# What Is Computer Vision

Let's revisit our example from the previous chapter: kicking a ball. As we have seen, this involves multiple tasks that our brains can do in a split second. The extraction of what is meaningful information from an image input is at the core of computer vision. But what is computer vision?

## Definition

Computer vision is the science and technology of making machines see. It involves the development of theoretical and algorithmic methods to acquire, process, analyze, and understand visual data, and to use this information to produce meaningful representations, descriptions, and interpretations of the world. (*Forsyth & Ponce, Computer Vision: A Modern Approach*)

> ## Deep learning and the computer vision renaissance

Reviewer 2: ```suggestion
## Deep Learning and the Computer Vision Renaissance
```

The evolution of computer vision has been marked by a series of incremental advancements in and across its interdisciplinary fields, where each step forward gave rise to breakthrough algorithms, hardware, and data, giving it more power and flexibility. One such leap was the jump to the widespread use of deep learning methods.

Initially, to extract and learn information in an image, you extract features through image-preprocessing techniques (chapter 3). Once you have a group of features describing your image, you use a classical machine learning algorithm on your dataset of features. It is a strategy that already simplifies things from the hard-coded rules, but it still relies on domain knowledge and exhaustive feature engineering. A more state-of-the-art approach arises when deep learning methods and large datasets meet. Deep learning (DL) allows machines to automatically learn complex features from the raw data. This paradigm shift allowed us to build more adaptive and sophisticated models, causing a renaissance in the field.

> ## Interdisciplinary with other fields + Image Understanding

Reviewer 2: ```suggestion
## Interdisciplinary Relations and Image Understanding
```

Just as it is hard to draw a line that separates artificial intelligence and computer vision,

> it is also hard to separate computer vision from its neighbouring fields. Take image preprocessing and analysis

Reviewer 2: There's a newline here, can you remove? From this paragraph there's multiple of them

Combining a “preprocessing” part before moving on to a more robust model is usual.
On the opposite side, sometimes, the layers of a neural network automatically perform tasks like these,
eliminating the need for explicit preprocessing. Image analysis might act as a first exploratory data analysis
step for those familiar with data science. Lastly, classical image analysis methods can also be used for
data augmentation to improve the quality and diversity of training data for computer vision models.

## Computer Vision Tasks Overview

We have seen before that computer vision is really hard for computers because they have no previous knowledge of the world. In our example, we start knowing what a ball is, how to track its movement, how objects usually move in space, how to estimate when the ball will reach us, where your foot is, how a foot moves, and how to estimate how much force you need to hit the ball. If we were to break this down into specific computer vision tasks, we would have:
  - Scene Recognition 
  - Object Recognition
  - Object Detection
  - Segmentation (instance, semantic)
  - Tracking 
  - Dynamic Environment Adaptation
  - Path Planning

> You will read more about the core tasks of computer vision in the [CV TASKS] chapter.

Reviewer 2: ```suggestion
You will read more about the core tasks of computer vision in the Computer Vision Tasks chapter.
```
I think you can put a link here once they're merged

But there are many more tasks that computer vision can do! Here is a non-exhaustive list:

  - Image Captioning
  - Image Classification
  - Image Description
  - Anomaly Detection
  - Image Generation
  - Image Restoration
  - Autonomous Exploration
  - Localization

> #TODO : add links to specific chapters that discuss some of these tasks

Reviewer 2: ```suggestion
```
We can remove for now to get this merged first
